main.py

```diff
@@ -0,0 +1,147 @@
+		print("6. Quit")
+		action = input("> ")
+
+		try:
+			if int(action) > 0 and int(action) < 7:
+				if action == "1":
+					# take user input and use to create new User()
+					print("\nEnter full name")
+					full_name = input("> ")
+					print("")
+					print("Enter screen name")
+					screen_name = input("> ")
+					user = UserUtility.create_user(full_name, screen_name)
+					# write User() info to CSV file
+					CSV.write_user_to_csv_file(user, "users.csv")
+					# set current user property
+					self.current_user = user
+					# set users property from users CSV file
+					self.users = CSV.get_users_from_csv_file("users.csv")
+					print("Welcome, {}!".format(self.current_user.screen_name))
+
+				elif action == "2":
+					# make sure there are users to choose from
+					try:
+						self.current_user = UserUtility.select_a_user(self.users, self.current_user)
+						print("Welcome, {}!".format(self.current_user.screen_name))
+					except AttributeError:
+						print("")
+						print("No users are registered.")
+
+				elif action == "3":
+					# display and select conversation thread
+					selected_convo_id = ChirpsUtility.view_chirps(
+						self.public_conversations,
+						self.private_conversations,
+						self.chirps,
+						self.current_user
+					)
+					# if user selected a public convo
+					if selected_convo_id[1] == "public":
+						# get selected conversation
+						chirp_list = self.public_conversations[selected_convo_id[0]]
+						# show all/add replies to selected convo
+						updated_convo = ConvoUtility.show_public_reply_menu(
+								self.chirps,
+								chirp_list,
+								self.current_user.screen_name
+								)
+						# add updated convo to public convo property
+						self.public_conversations[selected_convo_id[0]] = updated_convo
+						# write updated convo to CSV file
+						CSV.write_updated_convos_to_csv_file(self.public_conversations, "public_convos.csv")
+						# set public convo property from CSV file
+						self.public_conversations = CSV.get_convos_from_csv_file("public_convos.csv")
+						# set chirps property from CSV file
+						self.chirps = CSV.get_chirps_from_csv_file("chirps.csv")
+					# if user selected private convo
+					elif selected_convo_id[1] == "private":
+						chirp_list = self.private_conversations[selected_convo_id[0]]
+						# show all replies in private convo and add private reply
+						updated_convo = ConvoUtility.show_private_reply_menu(
+								self.chirps,
+								chirp_list,
+								self.current_user.screen_name
+								)
+						self.private_conversations[selected_convo_id[0]] = updated_convo
+						CSV.write_updated_convos_to_csv_file(self.private_conversations, "private_convos.csv")
+						self.private_conversations = CSV.get_convos_from_csv_file("private_convos.csv")
+						self.chirps = CSV.get_chirps_from_csv_file("chirps.csv")
+
+				elif action == "4":
+					# create new PublicChirp()
+					public_chirp = ChirpsUtility.new_public_chirp(self.current_user.screen_name)
```

Comment:
While your `ChirpsUtility.new_public_chirp` method has handling for a user not being logged in, this method argument attempts to get the `screen_name` off of `self.current_user`, which without a logged in user is an empty dictionary, and results in `AttributeError: 'dict' object has no attribute 'screen_name'` and a crash.